3ds Max/Max2Babylon/Forms/AnimationGroupControl.cs

```diff
@@ -167,6 +170,40 @@ private void confirmButton_Click(object sender, EventArgs e)
             ConfirmPressed?.Invoke(confirmedInfo);
         }
 
+        private AnimationGroup CalculateEndFrameFromAnimationGroupNodes(AnimationGroup animationGroup)
+        {
+            int endFrame = 0;
+            foreach (uint nodeHandle in currentInfo.NodeHandles)
+            {
+                IINode node = Loader.Core.GetINodeByHandle(nodeHandle);
+                if (node.IsAnimated && node.TMController!=null)
+                {
+                    int lastKey = 0;
+                    if (node.TMController.PositionController != null)
+                    {
+                        int posKeys = node.TMController.PositionController.NumKeys;
+                        lastKey = Math.Max(lastKey, node.TMController.PositionController.GetKeyTime(posKeys - 1));
+                    }
+
+                    if (node.TMController.RotationController!=null)
+                    {
+                        int rotKeys = node.TMController.RotationController.NumKeys;
+                        lastKey = Math.Max(lastKey, node.TMController.RotationController.GetKeyTime(rotKeys - 1));
+                    }
+
+                    if (node.TMController.ScaleController!=null)
```

Comment:
We're looking at posRotScale keys for the objects in our scene, but what about morph target weights? Or if the posRotScale of an object that we want animated is driven via IK or something that may not be keyframed?

Reply:
I don't really see this point, is the exporter able to export an IK bone without exporting the node that constraint the animation?

Reply:
Good point, something should be driving this that will be keyframed